WcaOnRails/app/models/incident.rb

```diff
@@ -38,4 +38,35 @@ def digest_sent_at_consistent
       errors.add(:digest_sent_at, "can't be set if incident is not resolved.")
     end
   end
+
+  def url
+    Rails.application.routes.url_helpers.incident_url(self)
+  end
+
+  DEFAULT_SERIALIZE_OPTIONS = {
+    only: ["id", "title", "public_summary"],
+    methods: ["url"],
+  }.freeze
+
+  def serializable_hash(options = nil)
+    json = super(DEFAULT_SERIALIZE_OPTIONS.merge(options || {}))
+    json.merge!(
+      class: self.class.to_s.downcase,
```

Comment:
Is this needed for the omnisearch, or anything in particular?

Reply:
Yes absolutely: omnisearch gets a bunch of different objects to display in the same select (users, competitions, regs, incidents, etc), this field let's the client now about which objects it's manipulating (for instance it's used [here](https://github.com/thewca/worldcubeassociation.org/pull/7208/files#diff-320e33e2efdf553ae279a78415ac32a4f30169fcf21f17a7af5b922d73526c5dR22) in this diff).

In retrospect we should probably have named the field `classString` or something to avoid the conflict with ruby's `class` method, but it's definitely too late now as it's propagated everywhere in public api endpoints :(